common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java

```diff
@@ -83,4 +112,49 @@ public void onBackPressed() {
             super.onBackPressed();
         }
     }
+
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+        adjustLayout();
+    }
+
+    private void adjustLayout(){
```

Comment:
Can these constraints be defined declaratively in the XML?

Something like:
```java
if (appIsSpannned) {
    setContentView(R.layout.spanned_authorization_activity);
} else {
    setContentView(R.layout.authorization_activity);
}
```

Reply:
Yes, but we'll need three files for each dual screen activities.
1. Not spanned.
2. Spanned - horizontal.
3. Spanned - vertical.

NOTE: after getting this feedback, I realized that the current solution can be made more reusable. Therefore, I made a change to do the following.

1. Created a 'dual_screen_layout.xml'. 
   - Internally, this contains the 'content' area, and the 'blank' area.

2. Moved adjustLayout() to DualScreenUtil.
  - This function now takes an activity that uses 'dual_screen_layout.xml'.
  - Based on the screen state, it will adjust the 'content' and the 'blank' area appropriately.

3. Any page that needs to support duo will have to do the following
 - call setContentView(R.layout.dual_screen_layout);
 - attach its 'content to be displayed' to the 'content' area of dual_screen_layout.xml.
 - call DualScreenUtil.adjustLayoutForDualScreenActivity().

This means I'll be able to reuse this with AuthorizationActivity (Broker/MSAL),  AuthenticationActivity (ADAL), and any other Broker pages that might have to support Duo in the future.

Reply:
@rpdome Overkill to define an `Activity` subclass that overrides `setContentView()` for this to support dual-screen?

Just riffing ideas

Reply:
That might make it cleaner. Let me try that.

Reply:
@iambmelt  done. Thanks for the suggestion.